### PR TITLE
HMSS-280 - Add Google SSO + deploy mobile app

### PR DESCRIPTION
**Pull request by DroppedHard**
TLDR - We need to switch to developer builds - otherwise the app will not work. This is required for Google SSO.

BIG CHANGE FOR US - PLEASE READ
Because the ONLY dependency that worked is [this one](https://react-native-google-signin.github.io) there is a big change to app development. This dependency requires a ready application. For android it was quite simple - because Android devices can install custom .apk files. With this I could simply build it with `eas build --platform android --profile development` code. STILL it is a development build, so it requires machine to run all JS code (as it will change during development). 

For iOS a developer account is required to install built apps. This is insanely bad, because now our current approach with Expo go is not working (because it is a RN library it needs to be built - tried to avoid that one REALLY bad - if someone finds a solution that is not insane we can manage). 

I used this tutorials to do this:
- [How to use a development build](https://www.youtube.com/watch?v=TJVc2D3LaPk)
- [Google Signin tutorial](https://www.youtube.com/watch?v=T-zTZn_xRBM)

For managing the [Google console](https://console.cloud.google.com/apis/credentials?authuser=6&hl=pl&inv=1&invt=AbliQw&project=health-monitoring-system-auth) and [eas builds](https://expo.dev/accounts/health-monitoring-system/projects/NaZdrowie/builds) - I used HealthMonitoringSystem2025 email. Everything should be available when you log in using our org email.

Additionally I fixes require cycles, some linter issues, and orher shenanigans.
EDIT - maybe a [free developer account](https://forums.developer.apple.com/forums/thread/742367) is sufficient?

### CODE REVIEW

**Comment by DroppedHard** [comment]
Google SSO will be added just for Android at the end
